### PR TITLE
Move onto new reactor architecture

### PR DESCRIPTION
**Pull request by dr-orlovsky**

### CODE REVIEW

File: radicle-node/src/wire/new.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+            }
+            Some(Io::Connect(id, addr)) => {
+                let NetAddr { host, port } = *addr;
+                let socket_addr = match host {
+                    HostAddr::Ip(ip) => net::SocketAddr::new(ip, addr.port()),
+                    HostAddr::Dns(_) => todo!(),
+                    _ => self.proxy,
+                };
+
+                self.inner.attempted(&socket_addr);
```
Comment (cloudhead) [inline radicle-node/src/wire/new.rs:245]: We shouldn't call this here, it should only be called if we are able to create the socket. I guess it's replaced by `connecting` anyway, which we didn't have before?

File: radicle-node/src/wire/new.rs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                    }
+                    Err(err) => {
+                        self.inner
+                            .disconnected(&id, &DisconnectReason::DialError(Arc::new(err)));
+                        return None;
+                    }
+                }
+            }
+            Some(Io::Disconnect(id, r)) => {
+                self.inner.disconnected(&id, &DisconnectReason::Protocol(r));
```
Comment (cloudhead) [inline radicle-node/src/wire/new.rs:264]: Same here we should only call this when something actually happened, and not in the same function that is reading from the inner service.

**Review by cloudhead — COMMENTED** [review]
Concept ACK

File: radicle-node/Cargo.toml [hunk 3]
```diff
@@ -7,14 +7,19 @@ edition = "2021"
 
 [features]
 test = ["radicle/test", "radicle-crypto/test", "qcheck"]
+new_arch = ["poll-reactor", "netservices"]
```
Comment (slack-coder) [inline radicle-node/Cargo.toml:10]: What about implementing it directly into the project in place of making it an optional feature?

Comment (cloudhead) [reply radicle-node/Cargo.toml:10]: It will be, but it's easier to review this way.

**Comment by dr-orlovsky** [comment]
I have added business logic dealing with all possible exceptions (disconnecting unconnected peers etc) to the new Wire.

@cloudhead pls review https://github.com/radicle-dev/heartwood/pull/143/commits/4ee66a2644e555d7d4d1facfc4e0650ca4fddf54

**Comment by dr-orlovsky** [comment]
Superceded by #162

**Pull request closed without merging**